### PR TITLE
[doc] Fix broken cross links in the documentation

### PR DESCRIPTION
**Pull request by createyourpersonalaccount**
The link was broken and it is now fixed.

Let me know if you are interested in fixing the links in ReFrame's documentation; I can take up the task.

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -101,7 +101,7 @@ A test must also define a validation function which is decorated with the :func:
 This function will be used to validate the test's output after it is finished.
 ReFrame, by default, makes no assumption about whether a test is successful or not;
 it is the test's responsibility to define its validation.
-The framework provides a rich set of `utility functions <deferrable_functions_reference>`__ that help matching patterns and extract values from the test's output.
+The framework provides a rich set of :doc:`utility functions <deferrable_functions_reference>`__ that help matching patterns and extract values from the test's output.
```
Comment (vkarak) [inline docs/tutorial.rst:104]:
```suggestion
The framework provides a rich set of :doc:`utility functions <deferrable_functions_reference>` that help matching patterns and extract values from the test's output.
```

**Review by vkarak — CHANGES_REQUESTED** [review]
Could you also rebase the PR and the commit on `master`?

File: docs/tutorial.rst (hunk 1 above)
Comment (createyourpersonalaccount) [reply docs/tutorial.rst:104]: @vkarak Yeah I can. Do you want me to see what other links are broken and fix them?

Comment (vkarak) [reply docs/tutorial.rst:104]: That would be welcome.

Comment (createyourpersonalaccount) [reply docs/tutorial.rst]: Okay I went ahead and did that.

File: docs/pipeline.rst [hunk 2]
```diff
@@ -45,7 +45,7 @@ The Setup Phase
 
 During this phase the test will be set up for the currently selected system partition and programming environment.
 The :attr:`current_partition` and :attr:`current_environ` test attributes will be set and the paths associated to this test case (stage, output and performance log directories) will be created.
-A `job descriptor <regression_test_api.html#reframe.core.pipeline.RegressionTest.job>`__ will also be created for the test case containing information about the job to be submitted later in the pipeline.
+A :attr:`job descriptor <reframe.core.pipeline.RegressionTest.job>` will also be created for the test case containing information about the job to be submitted later in the pipeline.
```
Comment (vkarak) [inline docs/pipeline.rst:48]: The problem with this syntax here is that "job descriptor" will be formatted in verbatim, which is not what we want. That's why we used the explicit link here. I will revert this change.

File: docs/tutorial.rst [hunk 3]
```diff
@@ -379,10 +379,10 @@ Let's look at some key elements of the configuration:
 * The :attr:`~config.systems.hostnames` option defines a set of hostname patterns which ReFrame will try to match against the current system's hostname.
   The first matching system will become the current system and ReFrame will load the corresponding configuration.
 * The :attr:`~config.systems.partitions.scheduler` partition option defines the job scheduler backend to use on this partition.
-  ReFrame supports many `job schedulers <config_reference.html#config.systems.partitions.scheduler>`__.
+  ReFrame supports many :class:`job schedulers <config.systems.partitions.scheduler>`.
```
Comment (vkarak) [inline docs/tutorial.rst:382]: Use the original formatting here (see comment above about the rationale).

File: docs/tutorial.rst [hunk 4]
```diff
@@ -379,10 +379,10 @@ Let's look at some key elements of the configuration:
...
   The ``local`` scheduler that we use here is the simplest one and it practically spawns a process executing the generated test script.
 * The :attr:`~config.systems.partitions.launcher` partition option defines the parallel launcher to use for spawning parallel programs.
-  ReFrame supports all the major `parallel launchers <config_reference.html#config.systems.partitions.launcher>`__.
+  ReFrame supports all the major :attr:`parallel launchers <config.systems.partitions.launcher>`.
```
Comment (vkarak) [inline docs/tutorial.rst:385]: Use the original formatting here (see comment above about the rationale).

File: docs/tutorial.rst [hunk 5]
```diff
@@ -980,13 +984,13 @@ Interacting with workload managers
 ==================================
 
 ReFrame integrates with many HPC workload managers (batch job schedulers), including Slurm, PBS Pro, Torque and others.
-The complete list of scheduler backend can be found `here <config_reference.html#config.systems.partitions.scheduler>`__.
+The complete list of scheduler backend can be found :attr:`here <config.systems.partitions.scheduler>`.
```
Comment (vkarak) [inline docs/tutorial.rst:987]: Use the original formatting here (see comment above about the rationale).

File: docs/tutorial.rst [hunk 6]
```diff
@@ -980,13 +984,13 @@ Interacting with workload managers
...
 Tests in ReFrame are scheduler-agnostic in that they do not need to include any scheduler-specific information.
 Instead, schedulers are associated to system partitions.
 Each system partition in the configuration file defines the scheduler backend to use along with any scheduler-specific options that are needed to grant access to the desired nodes.
 
 HPC systems also come with parallel program launchers which are responsible for launching parallel programs onto multiple nodes.
-ReFrame supports all major `parallel launchers <config_reference.html#config.systems.partitions.launcher>`__ and allows users to easily define their own custom ones.
+ReFrame supports all major :attr:`launchers <config.systems.partitions.launcher>` and allows users to easily define their own custom ones.
```
Comment (vkarak) [inline docs/tutorial.rst:993]: Use the original formatting here (see comment above about the rationale).

File: docs/tutorial.rst [hunk 7]
```diff
@@ -1122,7 +1126,7 @@ Defining extra scheduler resources
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 This method comprises two steps.
-First, we need to define a `resource <config_reference.html#config.systems.partitions.resources>`__ in the partition configuration:
+First, we need to define a :attr:`resource <config.systems.partitions.resources>` in the partition configuration:
```
Comment (vkarak) [inline docs/tutorial.rst:1129]: Use the original formatting here (see comment above about the rationale).

File: docs/tutorial.rst [hunk 8]
```diff
@@ -1177,7 +1181,7 @@ You can do that with the following hook:
        self.job.launcher = getlauncher('local')()
 
 The :func:`~reframe.core.backends.getlauncher` utility function returns the type the implements the launcher with the given name.
-The supported launcher names are those registered with the framework, i.e., all the names listed `here <config_reference.html#systems.partitions.launcher>`__ as well as any `user-registered <custom-launchers>`__ launcher.
+The supported launcher names are those registered with the framework, i.e., all the names listed :attr:`here <config.systems.partitions.launcher>` as well as any :ref:`user-registered <custom-launchers>` launcher.
```
Comment (vkarak) [inline docs/tutorial.rst:1184]: Use the original formatting here (see comment above about the rationale).

File: docs/tutorial.rst [hunk 9]
```diff
@@ -1213,7 +1217,7 @@ The ``job_scheduler_preamble`` contains the backend job scheduler directives tha
 The ``prepare_cmds`` are commands that can be emitted before the test environment commands.
 These can be specified with the :attr:`~config.systems.partitions.prepare_cmds` partition configuration option.
 The ``env_load_cmds`` are the necessary commands for setting up the environment of the test.
-These include any modules or environment variables set at the `system partition level <config_reference.html#system-partition-configuration>`__ or any `modules <regression_test_api.html#reframe.core.pipeline.RegressionTest.modules>`__ or `environment variables <regression_test_api.html#reframe.core.pipeline.RegressionTest.variables>`__ set at the test level.
+These include any modules or environment variables set at the :ref:`system partition level <system-partition-configuration>` or any :attr:`modules <reframe.core.pipeline.RegressionTest.modules>` or :attr:`environment variables <reframe.core.pipeline.RegressionTest.variables>` set at the test level.
```
Comment (vkarak) [inline docs/tutorial.rst:1220]: Use the original formatting for the `:attr:` links (see comment above about the rationale).

File: docs/tutorial.rst [hunk 10]
```diff
@@ -1875,7 +1881,7 @@ Let's walk briefly through the most important parts of its configuration:
 - The :attr:`~config.logging.handlers_perflog.format` specifies how the log record will be formatted.
   Each placeholder of the form ``%(placeholder)s`` is replaced by the actual value during runtime.
   All placeholders starting with ``check_`` refer to test attributes.
-  You can check the complete list of supported placeholders in the configuration `reference guide <config_reference.html#config.logging.handlers.format>`__.
+  You can check the complete list of supported placeholders in the configuration :attr:`reference guide <config.logging.handlers.format>`.
```
Comment (vkarak) [inline docs/tutorial.rst:1884]: Use the original formatting here (see comment above about the rationale).

**Review by vkarak — CHANGES_REQUESTED** [review]

File: docs/pipeline.rst (hunk 2 above)
Comment (createyourpersonalaccount) [reply docs/pipeline.rst]: @vkarak I wonder if there's a way to get what I want while keeping the style as a hyperlink?

Comment (vkarak) [reply docs/pipeline.rst]: I haven't found a way that's why we resorted to this solution, which is a bit verbose.

Comment (createyourpersonalaccount) [reply docs/pipeline.rst]: I'm just thinking that it's hardcoding the HTML format.

Comment (createyourpersonalaccount) [reply docs/pipeline.rst]: Okay sounds good, I'll do it soon.

**Comment by createyourpersonalaccount** [comment]
@vkarak done, you can merge

**Comment by vkarak** [comment]
I have pushed some improvement and style fixes to your branch, which are lost with your forced push. I will revert the links myself and force push.

**Review by vkarak — APPROVED** [review]
Lgtm now. Thanks!

**Pull request merged**